institutions-api/db/db.py

```diff
@@ -30,6 +30,20 @@ def _full_osg_id(short_id):
     """ Get the full osg-htc url of an institution based on its ID suffix """
     return f"{OSG_ID_PREFIX}{short_id}"
 
+def _get_unused_osg_id(session: Session):
+    """ Generate an unused OSG ID """
+    MAX_TRIES = 1000 # Give up after hitting x collisions in a row
+    ID_LENGTH = 12
+
+    # TODO actually guaranteeing uniqueness here might be a bit overkill based on ID length
+    all_ids = set(session.scalars(select(Institution.topology_identifier)).all())
+    for _ in range(MAX_TRIES):
+        short_id = ''.join([choice(ascii_lowercase + digits) for _ in range(ID_LENGTH)])
+        next_id = f"{OSG_ID_PREFIX}{short_id}"
+        if not next_id in all_ids:
+            return next_id
+        raise HTTPException(500, "Unable to generate new unique ID")
```

Comment:
```suggestion
    raise HTTPException(500, "Unable to generate new unique ID")
```
I think you meant this to be after the for loop